zerver/views/message_fetch.py

```diff
@@ -72,9 +72,19 @@ def get_search_fields(
     content_matches: Iterable[Tuple[int, int]],
     topic_matches: Iterable[Tuple[int, int]],
 ) -> Dict[str, str]:
+    # Following replacement is necessary because SQL-style HTML escaping of
+    # apostrophe (used to obtain `topic_matches`) differs from the django utils
+    # or any other library use to escape the apostrophe ('). This leads
```

Comment:
There is no “SQL-style HTML escaping”. The `func.escape_html` used in `zerver/lib/narrow.py` is a reference to a custom function that we defined ourselves.
  https://github.com/zulip/zulip/blob/f45a6a6d995c0c935491734153e245201a5ed207/zerver/migrations/0001_initial.py#L61-L63

“Any other library” is misleading—there are many other libraries that do escaping in many different ways. Django itself switched from `&#39;` to `&#x27;` [in Django 3.0](https://docs.djangoproject.com/en/3.0/releases/3.0/#miscellaneous).

Reply:
@akshatdalton can you reply with how you resolved this and the other comments about escaping strategy?

Reply:
I updated my [commit message](https://github.com/zulip/zulip/pull/25818/commits/b6d550368f47a6a278020b5a86c57c1ff6c9cc3c).

```
To fix this SQL query is changed to return the HTML-escaped
topic name generated by the function `func.escape_html`.
```

Reply:
I've also added a [comment block](https://github.com/zulip/zulip/pull/25818/files#diff-55c7496b687a8de1d5046bcea7ed5f9f78b50f2db9bdfe87df31fec3b6ca125bR961-R977) in the code.